### PR TITLE
Leashable API

### PR DESCRIPTION
**Pull request by Lulu13022002**
Expand a bit the api to support leashable boat.

### CODE REVIEW

File: patches/api/0480-Leashable-API.patch [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
++    boolean isLeashed();
++
++    /**
++     * Gets the entity that is currently leading this entity.
++     *
++     * @return the entity holding the leash
++     * @throws IllegalStateException if not currently leashed
++     */
++    @NotNull
++    Entity getLeashHolder() throws IllegalStateException;
```
Comment (lynxplay) [inline patches/api/0480-Leashable-API.patch]: What is the gain throwing ISE over null?

Comment (Lulu13022002) [reply patches/api/0480-Leashable-API.patch]: This is to keep the compatibility with the old method.

**Comment by lynxplay** [comment]
Do you think it is worth merging with `Expand EntityUnleashEvent` into a general "improved leash API" patch?
Don't have strong feelings either way, just found it while reviewing.

**Comment by Lulu13022002** [comment]
I would rather not. There's already a Shearable and Bucketable patch that exists which seems cleaner.

**Review by lynxplay — APPROVED** [review]

**Review by Machine-Maker — APPROVED** [review]

**Pull request merged**